File: examples/python/aws-cdk/stacks/storage/stacks.py
from dataclasses import dataclass

from cdk_organizer.aws.stack_group import StackGroup
from stacks.config import EnvironmentStackGroupConfig
from templates.s3_stack import S3BucketStack, S3BucketStackParameters


@dataclass
class StorageStackGroupConfig(EnvironmentStackGroupConfig):
    bucket_name: str


class StorageStackGroup(StackGroup[StorageStackGroupConfig]):
    def _load_stacks(self) -> None:
        self.bucket = S3BucketStack(
            self.app,
            self.get_stack_name("bucket"),
            stack_group=self,
            parameters=S3BucketStackParameters(
                base_bucket_name=self.data.bucket_name
            )
        ).bucket

File: examples/typescript/aws-cdk/stacks/storage/stacks.ts
import { StackGroup } from '@awslv/aws-cdk-organizer';
import { BaseConfig } from '@awslv/cdk-organizer-core';
import { Bucket } from 'aws-cdk-lib/aws-s3';
import { S3Stack } from '../../templates/s3-stack';

type StorageStackGroupProps = {
  bucketName: string;
} & BaseConfig;

export class StorageStackGroup extends StackGroup<StorageStackGroupProps> {
  public bucket!: Bucket;

  override async loadStacks(): Promise<void> {
    this.bucket = new S3Stack(this, this.app, this.getStackName('bucket'), {
      bucketName: this.data.bucketName,
    }).bucket;
  }
}
